### PR TITLE
[RLlib] Add gradient logging as default.

### PR DESCRIPTION
**Pull request by simonsays1980**
## Why are these changes needed?

Gradients logging is highly important for training neural networks. It enables users to monitor training performance and health. Right now, a user can only monitor gradients when she uses clipping and then also only, if the `"global_norm"` is used for clipping. This PR instead proposes gradients logging as a default option in RLlib.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/algorithm_config.py [hunk 1]
```diff
@@ -2979,6 +2981,9 @@ def debugging(
                 (this is also printed out once at startup at the INFO level).
             log_sys_usage: Log system resource metrics to results. This requires
                 `psutil` to be installed for sys stats, and `gputil` for GPU metrics.
+            log_gradients: Log gradients to results. If this is `True` the global norm
```
Comment (sven1977) [inline rllib/algorithms/algorithm_config.py:2984]: see above: move to `reporting()`.

File: rllib/algorithms/algorithm_config.py [hunk 2]
```diff
@@ -3004,6 +3009,8 @@ def debugging(
             self.log_level = log_level
         if log_sys_usage is not NotProvided:
             self.log_sys_usage = log_sys_usage
+        if log_gradients is not NotProvided:
+            self.log_gradients = NotProvided
```
Comment (sven1977) [inline rllib/algorithms/algorithm_config.py:3013]: see above: move to `reporting()`.

File: rllib/core/learner/learner.py [hunk 3]
```diff
@@ -540,7 +540,7 @@ def postprocess_gradients_for_module(
         """
         postprocessed_grads = {}
 
-        if config.grad_clip is None:
+        if config.grad_clip is None or not config.log_gradients:
```
Comment (sven1977) [inline rllib/core/learner/learner.py:543]: This is wrong, correct? So if someone switches off `config.log_gradients`, but still wants to grad-clip, then we DON'T grad-clip anymore :(

Comment (sven1977) [reply rllib/core/learner/learner.py:543]: I think it has to be `and`.

**Comment by sven1977** [comment]
This is great! Left some questions/nits. ...

File: rllib/algorithms/algorithm_config.py [hunk 4]
```diff
@@ -2961,6 +2962,7 @@ def debugging(
         logger_config: Optional[dict] = NotProvided,
         log_level: Optional[str] = NotProvided,
         log_sys_usage: Optional[bool] = NotProvided,
+        log_gradients: Optional[bool] = NotProvided,
```
[the remark this answers is not in the document]
Comment (simonsays1980) [reply rllib/algorithms/algorithm_config.py:2965]: Good point! Will do.

File: rllib/core/learner/learner.py (hunk 3 above)
Comment (simonsays1980) [reply rllib/core/learner/learner.py:543]: Great catch! It should :)

File: rllib/algorithms/algorithm_config.py [hunk 5]
```diff
@@ -500,6 +500,7 @@ def __init__(self, algo_class: Optional[type] = None):
         self.logger_config = None
         self.log_level = "WARN"
         self.log_sys_usage = True
+        self.log_gradients = True
```
Comment (sven1977) [inline rllib/algorithms/algorithm_config.py:503]: nit nit: Can you move this to the default-value block, that's headered with `reporting()`. Keeps the c'tor structured.

**Review by sven1977 — APPROVED** [review]
LGTM. Just please move the default value in the AlgorithmConfig constructor into the block commented as `# .reporting()`.

File: rllib/algorithms/algorithm_config.py (hunk 5 above)
Comment (simonsays1980) [reply rllib/algorithms/algorithm_config.py:503]: Great catch! Moved it to reporting defaults.
